spec/models/gestionnaire_spec.rb

```diff
@@ -78,4 +79,105 @@
+      let!(:follow_commentaire_groupe_gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire) { create(:follow_commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, gestionnaire: gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire, sender: administrateur, commentaire_seen_at: 1.day.ago) }
+
+      it do
+        expect(gestionnaire.unread_commentaires?(groupe_gestionnaire)).to eq false
+        expect(gestionnaire_unread_commentaire_cause_never_seen.unread_commentaires?(groupe_gestionnaire)).to eq true
+        expect(gestionnaire_unread_commentaire_cause_seen_at_before_last_commentaire.unread_commentaires?(groupe_gestionnaire)).to eq true
+      end
+    end
+  end
+
+  describe "#commentaire_seen_at" do
+    context "when already seen commentaire" do
+      let(:gestionnaire) { create(:gestionnaire) }
+      let(:administrateur) { create(:administrateur) }
+      let(:groupe_gestionnaire) { create(:groupe_gestionnaire, gestionnaires: [gestionnaire]) }
+      let!(:commentaire_groupe_gestionnaire) { create(:commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, sender: administrateur, created_at: 12.hours.ago) }
+      let!(:follow_commentaire_groupe_gestionnaire) { create(:follow_commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, gestionnaire: gestionnaire, sender: administrateur, commentaire_seen_at: Time.zone.now) }
+
+      it { expect(gestionnaire.commentaire_seen_at(groupe_gestionnaire, administrateur)).not_to eq nil }
+    end
+
+    context "when never seen commentaire" do
+      let(:gestionnaire) { create(:gestionnaire) }
+      let(:administrateur) { create(:administrateur) }
+      let(:groupe_gestionnaire) { create(:groupe_gestionnaire, gestionnaires: [gestionnaire]) }
+      let!(:commentaire_groupe_gestionnaire) { create(:commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, sender: administrateur, created_at: 12.hours.ago) }
+
+      it { expect(gestionnaire.commentaire_seen_at(groupe_gestionnaire, administrateur)).to eq nil }
+    end
+  end
+
+  describe "#mark_commentaire_as_seen" do
+    context "when already seen commentaire" do
+      let(:now) { Time.zone.now.beginning_of_minute }
+      let(:gestionnaire) { create(:gestionnaire) }
+      let(:administrateur) { create(:administrateur) }
+      let(:groupe_gestionnaire) { create(:groupe_gestionnaire, gestionnaires: [gestionnaire]) }
+      let!(:commentaire_groupe_gestionnaire) { create(:commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, sender: administrateur, created_at: 12.hours.ago) }
+      let!(:follow_commentaire_groupe_gestionnaire) { create(:follow_commentaire_groupe_gestionnaire, groupe_gestionnaire: groupe_gestionnaire, gestionnaire: gestionnaire, sender: administrateur, commentaire_seen_at: 12.hours.ago) }
+
+      before do
+        Timecop.freeze(now) do
```

Comment:
```suggestion
        travel_to(now) do
```
(je sais qu'il y a encore beaucoup d'occurences de timecop dans notre codebase, mais les helpers de rails sont suffisants)

Reply:
done